vm/proxyapp/proxyappclient.go

```diff
@@ -0,0 +1,391 @@
+			Workdir: workdir,
+			Index:   index},
+		&reply)
+	if err != nil {
+		return nil, fmt.Errorf("failed to proxy.Call(\"ProxyVM.CreateInstance\"): %w", err)
+	}
+
+	atomic.AddInt32(&proxy.instancesCreated, 1)
+	return &instance{
+		ProxyApp: proxy,
+		ID:       reply.ID,
+	}, nil
+}
+
+type instance struct {
+	*ProxyApp
+	ID string
+}
+
+// Copy copies a hostSrc file into VM and returns file name in VM.
+// nolint: dupl
+func (inst *instance) Copy(hostSrc string) (string, error) {
+	var reply proxyrpc.CopyResult
+	err := inst.ProxyApp.Call(
+		"ProxyVM.Copy",
+		proxyrpc.CopyParams{
+			ID:      inst.ID,
+			HostSrc: hostSrc,
+		},
+		&reply)
+	if err != nil {
+		return "", err
+	}
+
+	return reply.VMFileName, nil
+}
+
+// Forward sets up forwarding from within VM to the given tcp
+// port on the host and returns the address to use in VM.
+// nolint: dupl
+func (inst *instance) Forward(port int) (string, error) {
+	var reply proxyrpc.ForwardResult
+	err := inst.ProxyApp.Call(
+		"ProxyVM.Forward",
+		proxyrpc.ForwardParams{
+			ID:   inst.ID,
+			Port: port,
+		},
+		&reply)
+	if err != nil {
+		return "", err
+	}
+	return reply.ManagerAddress, nil
+}
+
+func buildMerger(names ...string) (*vmimpl.OutputMerger, []io.Writer) {
+	var wPipes []io.Writer
+	merger := vmimpl.NewOutputMerger(nil)
+	for _, name := range names {
+		rpipe, wpipe := io.Pipe()
+		wPipes = append(wPipes, wpipe)
+		merger.Add(name, rpipe)
+	}
+	return merger, wPipes
+}
+
+func (inst *instance) Run(
+	timeout time.Duration,
+	stop <-chan bool,
+	command string,
+) (<-chan []byte, <-chan error, error) {
+	merger, wPipes := buildMerger("stdout", "stderr", "console")
+	receivedStdoutChunks := wPipes[0]
+	receivedStderrChunks := wPipes[1]
+	receivedConsoleChunks := wPipes[2]
+	outc := merger.Output
+
+	var reply proxyrpc.RunStartReply
+	err := inst.ProxyApp.Call(
+		"ProxyVM.RunStart",
+		proxyrpc.RunStartParams{
+			ID:      inst.ID,
+			Command: command},
+		&reply)
+
+	if err != nil {
+		return nil, nil, fmt.Errorf("error calling ProxyVM.Run with command %v: %w", command, err)
+	}
+
+	runID := reply.RunID
+	errc := make(chan error, 1)
+	timeoutSignal := time.After(timeout)
+
+	go func() {
+		defer func() {
+			inst.ProxyApp.Call(
+				"ProxyVM.RunStop",
+				proxyrpc.RunStopParams{
+					ID:    inst.ID,
+					RunID: runID,
+				},
+				&proxyrpc.RunStopParams{})
+		}()
+
+		for {
+			var progress proxyrpc.RunReadProgressReply
+			readProgressCall := inst.ProxyApp.Go(
+				"ProxyVM.RunReadProgress",
+				proxyrpc.RunReadProgressParams{
+					ID:    inst.ID,
+					RunID: runID,
+				},
+				&progress,
+				nil)
+			select {
+			case <-readProgressCall.Done:
+				receivedStdoutChunks.Write([]byte(progress.StdoutChunk))
+				receivedStderrChunks.Write([]byte(progress.StderrChunk))
+				receivedConsoleChunks.Write([]byte(progress.ConsoleOutChunk))
+				if readProgressCall.Error != nil {
+					errc <- fmt.Errorf("error reading progress from %v:%v: %v",
```

Comment:
When we send an error to the channel, it will be treated as the tested kernel bug. If there is nothing in the output that looks like a kernel oops, it will classified as "lost connection to test machine".
I think we want to separate these errors (presumably the proxy bug, right?).
Write the error to one of received*Chunks as:
```
full error message
SYZFAIL: some-fixed-text
```
Also relevant for the next else if case.

Reply:
read calling side to get more details
don